server/src/main/java/org/spine3/server/storage/CommandStorage.java

```diff
@@ -72,31 +92,76 @@ public void store(Command command) {
      *
      * @param command a command to store
      * @param error an error occurred
+     * @throws IllegalStateException if the storage is closed
      */
     public void store(Command command, Error error) {
         checkNotClosed();
         checkNotDefault(error);
-        CommandId id = command.getContext().getCommandId();
+        CommandId id = getId(command);
         if (idToString(id).equals(EMPTY_ID)) {
             id = generateId();
         }
-        final CommandStorageRecord record = toStorageRecord(command)
-                .toBuilder()
-                .setStatus(CommandStatus.ERROR)
+        final CommandStorageRecord record = newCommandStorageRecordBuilder(command, ERROR)
                 .setError(error)
                 .setCommandId(idToString(id))
                 .build();
         write(id, record);
     }
 
+    /**
+     * Loads all commands with the given status.
+     *
+     * @param status a command status to search by
+     * @return commands with the given status
+     * @throws IllegalStateException if the storage is closed
+     */
+    public Iterator<Command> load(CommandStatus status) {
+        checkNotClosed();
+        final Iterator<CommandStorageRecord> recordIterator = read(status);
```

Comment:
We don't really `read` here. We get an iterator over the status. When we read, we load data from a media.